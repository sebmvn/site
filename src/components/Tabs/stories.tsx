import { Story, Meta } from '@storybook/react/types-6-0'
import Tab, { TabsProps } from '.'

export default {
  title: 'Tab',
  component: Tab,
  args: {
    components: [
      {
        title: 'Tab Component 1',
        content: <span>Component 1</span>
      },
      {
        title: 'Tab Component 2',
        content: <span>Component 2</span>
      },
      {
        title: 'Tab Component 3',
        content: <span>Component 3</span>
      }
    ]
  }
} as Meta

export const Default: Story<TabsProps> = (args) => <Tab {...args} />
