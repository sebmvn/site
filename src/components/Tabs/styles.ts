import styled, { css } from 'styled-components'

export const Wrapper = styled.main`
  ${({ theme }) => css`
    width: 100%;
    height: auto;
    min-height: 400px;
    padding: ${theme.spacings.small} 0;
    color: ${theme.colors.primary};
    border-radius: ${theme.border.radius};
  `}
`

export const Nav = styled.ul`
  ${({ theme }) => css`
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: space-between;

    margin: 0 auto ${theme.spacings.small};
    border: 2px solid ${theme.colors.primary};
    border-radius: ${theme.border.radius};
  `}
`

export const NavItem = styled.li`
  ${({ theme }) => css`
    width: 50%;
    padding: ${theme.spacings.xsmall};
    list-style: none;
    text-align: center;
    cursor: pointer;
    transition: all 0.7s;
    border-radius: 0;
    font-weight: bold;

    &:first-child() {
      border-bottom-left-radius: ${theme.border.radius};
      border-top-left-radius: ${theme.border.radius};
    }

    &:last-child() {
      border-bottom-right-radius: ${theme.border.radius};
      border-top-right-radius: ${theme.border.radius};
    }

    &:hover {
      background: ${theme.colors.primary};
      color: ${theme.colors.white};
    }

    &.active {
      background: ${theme.colors.secondary};
      color: ${theme.colors.primary};
    }
  `}
`

export const TabResult = styled.div``
