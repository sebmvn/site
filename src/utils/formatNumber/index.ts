export const formatPriceNotation = (value: string) => {
  value = value?.replace('.', '').replace(',', '').replace(/\D/g, '')

  if (isNaN(parseFloat(value))) {
    return ''
  }

  const options = { minimumFractionDigits: 2 }
  const result = new Intl.NumberFormat('pt-BR', options).format(
    parseFloat(value) / 100
  )

  return result
}

export const formatIntegerNotation = (value: string) => {
  return value?.replace('.', '').replace(',', '').replace(/\D/g, '')
}

export const toDecimal = (value: string) => {
  const formattedValue = formatIntegerNotation(value)
  return parseFloat(formattedValue!) / 100
}

export const toInteger = (value: string) => {
  const formattedValue = formatIntegerNotation(value)
  return parseInt(formattedValue!)
}
