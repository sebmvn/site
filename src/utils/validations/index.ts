import { CalendarType } from 'hooks/use-calc'
import Joi from 'joi'

const fieldsValidations = {
  initialValue: Joi.string().required().messages({
    'any.required': 'Campo obrigatório',
    'string.empty': 'Campo obrigatório',
    'string.min': 'tamanho minimo de 1 digito',
    'string.base': 'o tipo do dado precisa ser textual'
  }),
  monthlyValue: Joi.string().required().messages({
    'any.required': 'Campo obrigatório',
    'string.empty': 'Campo obrigatório',
    'string.min': 'tamanho minimo de 1 digito',
    'string.base': 'o tipo do dado precisa ser textual'
  }),

  interestRate: Joi.string().required().messages({
    'any.required': 'Campo obrigatório',
    'string.empty': 'Campo obrigatório',
    'string.min': 'tamanho minimo de 1 digito',
    'string.base': 'o tipo do dado precisa ser textual'
  }),
  period: Joi.string().required().messages({
    'any.required': 'Campo obrigatório',
    'string.empty': 'Campo obrigatório',
    'string.min': 'tamanho minimo de 1 digito',
    'string.base': 'o tipo do dado precisa ser textual'
  }),
  interestRateCalendar: Joi.string().valid('monthly', 'yearly').required(),
  periodCalendar: Joi.string().valid('monthly', 'yearly').required()
}

export type FieldErrors = {
  [key: string]: string
}

const getFieldErrors = (objError: Joi.ValidationResult) => {
  const errors: FieldErrors = {}

  if (objError.error) {
    objError.error.details.forEach((err) => {
      errors[err.path.join('.')] = err.message
    })
  }

  return errors
}

type compoundInterestCalculatorFields = {
  initialValue: string
  monthlyValue: string
  interestRate: string
  period: string
  interestRateCalendar: CalendarType
  periodCalendar: CalendarType
}
export const compoundInterestCalculatorValidate = (
  values: compoundInterestCalculatorFields
) => {
  const schema = Joi.object(fieldsValidations)

  return getFieldErrors(schema.validate(values, { abortEarly: false }))
}
