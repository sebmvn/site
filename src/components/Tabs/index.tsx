import { useState } from 'react'
import * as S from './styles'

export type TabsProps = {
  components: {
    title: string
    content: React.ReactNode
  }[]
}

const Tabs = ({ components }: TabsProps) => {
  const [activeTab, setActiveTab] = useState({ classname: 'tab-0', index: 0 })

  const handleTab = (classname: string, index: number) => {
    setActiveTab({ classname, index })
  }

  return (
    <S.Wrapper>
      <S.Nav>
        {components.map((component, index) => (
          <S.NavItem
            key={index}
            className={activeTab.classname === `tab-${index}` ? 'active' : ''}
            onClick={() => handleTab(`tab-${index}`, index)}
          >
            {component.title}
          </S.NavItem>
        ))}
      </S.Nav>

      <S.TabResult>{components[activeTab.index].content}</S.TabResult>
    </S.Wrapper>
  )
}

export default Tabs
