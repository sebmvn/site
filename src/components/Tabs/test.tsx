import { renderWithTheme } from 'utils/test-utils'

import Tabs from '.'

describe('<Tabs />', () => {
  const components = [
    {
      title: 'Tab Component 1',
      content: <span>Component 1</span>
    },
    {
      title: 'Tab Component 2',
      content: <span>Component 2</span>
    },
    {
      title: 'Tab Component 3',
      content: <span>Component 3</span>
    }
  ]
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Tabs components={components} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
