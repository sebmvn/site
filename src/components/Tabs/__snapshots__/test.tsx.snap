// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> should render the heading 1`] = `
.c0 {
  width: 100%;
  height: auto;
  min-height: 400px;
  padding: 2.4rem 0;
  color: #193241;
  border-radius: 0.5rem;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0 auto 2.4rem;
  border: 2px solid #193241;
  border-radius: 0.5rem;
}

.c2 {
  width: 50%;
  padding: 1.6rem;
  list-style: none;
  text-align: center;
  cursor: pointer;
  -webkit-transition: all 0.7s;
  transition: all 0.7s;
  border-radius: 0;
  font-weight: bold;
}

.c2:first-child() {
  border-bottom-left-radius: 0.5rem;
  border-top-left-radius: 0.5rem;
}

.c2:last-child() {
  border-bottom-right-radius: 0.5rem;
  border-top-right-radius: 0.5rem;
}

.c2:hover {
  background: #193241;
  color: #FFFFFF;
}

.c2.active {
  background: #0ADC7D;
  color: #193241;
}

<main
  class="c0"
>
  <ul
    class="c1"
  >
    <li
      class="c2 active"
    >
      Tab Component 1
    </li>
    <li
      class="c2"
    >
      Tab Component 2
    </li>
    <li
      class="c2"
    >
      Tab Component 3
    </li>
  </ul>
  <div
    class=""
  >
    <span>
      Component 1
    </span>
  </div>
</main>
`;
